Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs

```diff
@@ -7,41 +7,68 @@ internal sealed class FontTab : HtmlTab
 {
 	public byte[] Data { get; }
 
+	public string Source { get; }
+
 	public string? FileName { get; }
 
+	public string? MimeType { get; }
+
 	public override string DisplayName => Localization.AssetTabFont;
 
 	public override string HtmlName => "font";
 
-	public override bool Enabled => Data.Length > 0;
+	public override bool Enabled => !string.IsNullOrEmpty(Source);
 
 	public FontTab(IUnityObjectBase asset)
 	{
-		if (asset is IFont font)
-		{
-			Data = font.FontData;
-			if (Data is { Length: >= 4 })
-			{
-				FileName = $"{font.GetBestName()}.{(Data[0], Data[1], Data[2], Data[3]) switch
-				{
-					(0x4F, 0x54, 0x54, 0x4F) => "otf",
-					(0x00, 0x01, 0x00, 0x00) => "ttf",
-					(0x74, 0x74, 0x63, 0x66) => "ttc",
-					_ => "",
-				}}";
-			}
-		}
-		else
-		{
-			Data = [];
-		}
+		Source = TryDecode(asset, out byte[] data, out string? fileName, out string? mimeType);
+		Data = data;
+		FileName = fileName;
+		MimeType = mimeType;
 	}
 
 	public override void Write(TextWriter writer)
 	{
 		using (new Div(writer).WithClass("text-center").End())
 		{
-			DataSaveButton.Write(writer, FileName, Data);
+			new H1(writer).WithStyle($"font-family: {FileName}").Close("Preview Font (0, 1, 2, 3, 4, 5, 6, 7, 8, 9)");
+
+			DataSaveButton.Write(writer, $"{FileName}.{MimeType}", Data);
+
+			new Script(writer).Close(
+				$$"""
+				  const fontFace = new FontFace(`{{FileName}}`, `url({{Source}})`);
+				  document.fonts.add(fontFace);
+				  fontFace.load().then().catch(function(error) {
+				    console.error(`Font loading failed: ${error}`);
+				  });
+				  """);
 		}
 	}
+
+	private static string TryDecode(IUnityObjectBase asset, out byte[] data, out string? fileName, out string? mimeType)
```

Comment:
Methods with this design pattern should always have a boolean return type.

Reply:
And this part is similar to the code:
https://github.com/AssetRipper/AssetRipper/blob/6771232426dc4a38e621ccaf034d916e11a2d79b/Source/AssetRipper.GUI.Web/Pages/Assets/AudioTab.cs#L39-L46

Reply:
Of course, I can put this part of the code like AudioClipDecoder

Reply:
That code doesn't have any out parameters, which makes it different. In this case, we want a Boolean return. The current return can be made into another out parameter.